xmtp_ecies/src/signed_payload.rs

```diff
@@ -0,0 +1,170 @@
+use libsecp256k1::{
+    sign as secp_sign, Error as SecpError, Message, SecretKey, Signature as SecpSignature,
+};
+use prost::Message as ProstMessage;
+use xmtp_proto::xmtp::message_contents::{
+    ecies_message::Version as EciesVersion, signature::EcdsaCompact, signature::Union,
+    EciesMessage, Signature, SignedPayload,
+};
+
+use crate::{
+    ecies::{decrypt_bytes, encrypt_bytes},
+    hash::sha3_256,
+};
+
+pub fn encrypt_message(
+    public_key: &[u8],
+    private_key: &[u8],
```

Comment:
Would it be worth specifying that these must be secp256k1 keys?

Reply:
Great call out. Will add